controllers/amphoracontroller_controller.go

```diff
@@ -106,28 +106,85 @@ func (r *OctaviaAmphoraControllerReconciler) Reconcile(ctx context.Context, req
 			// For additional cleanup logic, use finalizers. Return and don't requeue.
 			return ctrl.Result{}, nil
 		}
-		// Error reading the object, requeue the request
+		// Error reading the object - requeue the request.
+		Log.Error(err, fmt.Sprintf("could not fetch instance %s", instance.Name))
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.Conditions == nil {
-		// Setup the initial conditions
+	helper, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		Log,
+	)
+	if err != nil {
+		Log.Error(err, fmt.Sprintf("could not instantiate helper for instance %s", instance.Name))
+		return ctrl.Result{}, err
+	}
+
+	// initialize status if Conditions is nil, but do not reset if it already
+	// exists
+	isNewInstance := instance.Status.Conditions == nil
+	if isNewInstance {
 		instance.Status.Conditions = condition.Conditions{}
-		cl := condition.CreateList(
-			condition.UnknownCondition(condition.ExposeServiceReadyCondition, condition.InitReason, condition.ExposeServiceReadyInitMessage),
-			condition.UnknownCondition(condition.ServiceConfigReadyCondition, condition.InitReason, condition.ServiceConfigReadyInitMessage),
-			condition.UnknownCondition(condition.InputReadyCondition, condition.InitReason, condition.InputReadyInitMessage),
-			condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.DeploymentReadyInitMessage),
-			condition.UnknownCondition(condition.NetworkAttachmentsReadyCondition, condition.InitReason, condition.NetworkAttachmentsReadyInitMessage),
-			condition.UnknownCondition(condition.TLSInputReadyCondition, condition.InitReason, condition.InputReadyInitMessage),
-		)
-		// TODO(beagles): what other conditions?
-		instance.Status.Conditions.Init(&cl)
+	}
+
+	// Save a copy of the condtions so that we can restore the LastTransitionTime
+	// when a condition's state doesn't change.
+	savedConditions := instance.Status.Conditions.DeepCopy()
+
+	// Always patch the instance status when exiting this function so we can
+	// persist any changes.
+	defer func() {
+		condition.RestoreLastTransitionTimes(
+			&instance.Status.Conditions, savedConditions)
+		if instance.Status.Conditions.IsUnknown(condition.ReadyCondition) {
+			instance.Status.Conditions.Set(
+				instance.Status.Conditions.Mirror(condition.ReadyCondition))
+		}
+
+		if err := helper.SetAfter(instance); err != nil {
```

Comment:
@abays @beagles I kept the old code here which consists on L147-L152.
All the other operators usually patch the instance with the following:

```golang

...
err := helper.PatchInstance(ctx, instance)
    if err != nil {
	_err = err
	return
}
```
I'm wondering if we want to move to that approach or there's a particular reason to look for the changes in the `status` struct before calling `Patch` (that should be covered in [1])

[1] https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/helper/helper.go#L193

Reply:
I think we should use the same pattern employed in other operators

Reply:
Setting to resolved. Seems this was done.